gh-120170: Exclude __mp_main__ in C version of whichmodule()

li-dan opened the pull request:
Importing multiprocessing adds an alias to `__main__` named `__mp_main__`. In #23403, the Python version of whichmodule() was fixed to exclude `__mp_main__`. Apply the same fix to the C version of the function.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->


<!-- gh-issue-number: gh-120170 -->
* Issue: gh-120170
<!-- /gh-issue-number -->

lysnikolaou commented on Misc/NEWS.d/next/Library/2024-06-06-17-49-07.gh-issue-120170.DUxhmT.rst:
```diff
@@ -0,0 +1,3 @@
+We fixed an issue in the :mod:`!_pickle` extension module in which importing
```
```suggestion
Fix an issue in the :mod:`!_pickle` extension module in which importing
```

lysnikolaou reviewed (approved):
Thanks for the PR @li-dan! I'm going to approve this, since it seems straightforward enough. However, it'd be best if someone else was to give it a quick look.

Also, should @li-dan re-sign the CLA? What's the process here? cc @ambv

li-dan committed:
gh-120170: Exclude __mp_main__ in C version of whichmodule()

Importing multiprocessing adds an alias to __main__ named __mp_main__.
In #23403, the Python version of whichmodule() was fixed to exclude
__mp_main__. Apply the same fix to the C version of the function.

lysnikolaou committed:
Merge branch 'main' into fix_120170
